kubernetes-server-mock/src/main/java/io/fabric8/kubernetes/client/server/mock/KubernetesAttributesExtractor.java

```diff
@@ -47,16 +51,29 @@
 
   protected static final Pattern PATTERN = Pattern.compile(API_GROUP + VERSION_GROUP + NAMESPACE_GROUP + KIND_GROUP + NAME_GROUP + END_GROUP);
 
+  private static final String LABEL_KEY_PREFIX = "labels:";
+  private static final String KEY_GROUP = "(?<key>[a-zA-Z0-9-_./]+)";
+  // Matches a==b and a=b but not a!=b.
+  private static final String EQUALITY_GROUP = "(==|(?<!!)=)";
+  private static final String VALUE_GROUP = "(?<value>[a-zA-Z0-9-_.]+)";
+  private static final Pattern LABEL_REQUIREMENT_EQUALITY = Pattern.compile(KEY_GROUP + EQUALITY_GROUP + VALUE_GROUP);
+
+  private HttpUrl parseUrlFromPathAndQuery(String s) {
```

Comment:
Is it possible to get rid of this hard coded string?

Reply:
Apart from what @rohanKanojia suggests to have this `localhost` as constant I will do next things in this case. I know they might sounds weird but I have done this in several projects and works pretty well in terms of readable code.

. Create a `private static final String PROTOCOL = http`
. Create a `private static final String HOST = localhost`
. Check if `s` starts with `/` or not and in case of not, append it.
. Do `HttpUrl.parse(String.format("%s://%s%s", PROTOCOL, HOST, s))`

if `s` can contain the port as well, split the method between `int port, String path` it is always a better practice to not merge concepts.

Reply:
Done, thanks!